addons/im_livechat/static/src/core/common/thread_patch.js

```diff
@@ -0,0 +1,14 @@
+import { Thread } from "@mail/core/common/thread";
+
+import { patch } from "@web/core/utils/patch";
+
+patch(Thread.prototype, {
+    get showVisitorDisconnected() {
+        return (
+            this.props.thread.channel_type === "livechat" &&
+            this.store.self.isInternalUser &&
+            this.props.thread.livechat_active &&
+            this.props.thread.livechatVisitorMember?.persona.im_status != "online"
+        );
+    },
+});
```

Comment:
Visitor should be non-online for some time, otherwise any page load with slow internet connection shows disconnected, which can give wrong impression to livechat operator to end the conversation.

I'd say 1 or 2 minutes at the very least, taking into account loss of connection + failed retry may take 30 seconds for another retry.

Reply:
> I'll repeat what I suggested on the task but I really think we should display since how long the visitor is offline to let the operator decide what is "long enough" rather than pick an arbitrary value.

We need a threshold for time since becoming offline before showing this dynamically on UI at the very least, to prevent flickers from website page navigation of the visitor. This could simply be a few seconds timeout in operator JS code when there's an online -> offline transition (page reload immediately shows offline, which is not a problem).

Reply:
We already have a debounce in im status handling code to avoid flicker of standard im status display.

Reply:
> We already have a debounce in im status handling code to avoid flicker of standard im status display.

Probably not long enough them, with current branch and testing locally the flicker is there when page reloading.